cmd/create/admin/cmd.go

```diff
@@ -176,8 +188,25 @@ func run(cmd *cobra.Command, _ []string) {
 	r.Reporter.Infof("It may take several minutes for this access to become active.")
 }
 
-func FindExistingClusterAdminIDP(cluster *cmv1.Cluster, r *rosa.Runtime) (
-	htpasswdIDP *cmv1.IdentityProvider, userList *cmv1.HTPasswdUserList) {
+// find the htpasswd idp "cluster-admin"
+func FindClusterAdminIDP(cluster *cmv1.Cluster, r *rosa.Runtime) *cmv1.IdentityProvider {
+	idps, err := r.OCMClient.GetIdentityProviders(cluster.ID())
+	if err != nil {
+		r.Reporter.Errorf("Failed to get identity providers for cluster '%s': %v", r.ClusterKey, err)
+		os.Exit(1)
```

Comment:
@marcolan018 I'd personally suggest returning the  `error` from this function rather than `os.Exit`, then you can test the error flow as well.

Reply:
changed 
@robpblake please help to re-review the latest commits, thanks